### ISSUE TITLE
Grid_guardCellMaskHook needed for Mode3

### ISSUE THREAD

**Issue by jaharris87** [ECP-Astro/FLASH5#642]
Currently, the `Grid_fillGuardCells` for AMReX uses `gr_makeMaskConsistent_gen`, which calls `Grid_guardCellMaskHook`.

Currently, this would just be a stub since the only non-stub `Grid_guardCellMaskHook.F90` exists in `Grid/GridMain/AMR/paramesh/paramesh4`.

I don't see anything in this file that is specific to Paramesh, though. I think it could just be moved up to `Grid/GridMain/AMR`.

**Comment by kweide** [comment]
I agree that `Grid_guardCellMaskHook.F90` could and should be moved as you suggest.

But note that the support for gc fill variable masking in our Amrex Grid implementation is rudimentary, and I suspect there is a bug anyway (need to test again); it is not in our regular test suites.

**Comment by jaharris87** [comment]
I'll soon have a test to add to the test suite (`StreamingSineWave` branch) that will test this.

**Comment by kweide** [comment]
I think #46 was where the potential bug I mentioned showed up. Someone should try that setup again. (I've been meaning to...)
